dppl/oneapi_interface.pyx

```diff
@@ -101,37 +94,66 @@ cdef class DpplRuntime:
     def _deactivate_current_queue (self):
         self.rt.deactivateCurrentQueue()
 
-    def get_current_queue (self):
-        ''' Returns the activated SYCL queue as a PyCapsule.
-        '''
-        cdef void* queue_ptr = NULL;
-        self.rt.getCurrentQueue(&queue_ptr);
-        return PyCapsule_New(queue_ptr, NULL, &delete_queue)
-
-#    def set_global_queue (self, device_ty, device_id):
-#        if device_ty == device_type.gpu:
-#            self.rt.setGlobalContextWithGPU(device_id)
-#        elif device_ty == device_type.cpu:
-#            self.rt.setGlobalContextWithCPU(device_id)
-#        else:
-#            e = UnsupportedDeviceTypeError("Device can only be cpu or gpu")
-#            raise e
-
     def dump (self):
         ''' Prints information about the Runtime object.
         '''
         return self.rt.dump()
 
-    def dump_queue (self, queue_cap):
+    def dump_queue_info (self, queue_cap):
         ''' Prints information about the SYCL queue object.
         '''
         if PyCapsule_IsValid(queue_cap, NULL):
             self.rt.dump_queue(PyCapsule_GetPointer(queue_cap, NULL))
         else:
             raise ValueError("Expected a PyCapsule encapsulating a SYCL queue")
 
-# Global runtime object
-runtime = DpplRuntime()
+    def get_current_queue (self):
+        ''' Returns the activated SYCL queue as a PyCapsule.
+        '''
+        cdef void* queue_ptr = NULL;
+        self.rt.getCurrentQueue(&queue_ptr);
+        return PyCapsule_New(queue_ptr, NULL, &delete_queue)
+
+    def get_num_platforms (self):
+        ''' Returns the number of available SYCL/OpenCL platforms.
+        '''
+        cdef size_t num_platforms = 0
+        self.rt.getNumPlatforms(&num_platforms)
+        return num_platforms
+
+    def set_default_queue (self, device_ty, device_id):
+        if device_ty == device_type.gpu:
+            self.rt.resetGlobalQueue(_device_type._GPU, device_id)
+        elif device_ty == device_type.cpu:
+            self.rt.resetGlobalQueue(_device_type._CPU, device_id)
+        else:
+            e = UnsupportedDeviceTypeError("Device can only be cpu or gpu")
+            raise e
+
+# thread-local storage
+from threading import local as threading_local
+
+# Initialize a thread local instance of _DpplRuntime
+_tls = threading_local()
+_tls._in_dppl_ctxt = False
```

Comment:
Not sure we ever discussed this scenario.  However, I think I agree that in a new thread, you go back to the default of not being in a context.

Reply:
I am removing the flag. Instead of the flag let me add a public getter that returns True/False based on if we are inside *any* context by checking the size of the stack of contexts.

Reply:
Made the change in 8318d44